tests/rptest/clients/rpk.py

```diff
@@ -312,14 +355,25 @@ def _check_stdout_success(self, output):
         if not status_line.endswith("OK"):
             raise RpkException(f"Bad status: '{status_line}'")
 
-    def sasl_allow_principal(self, principal, operations, resource,
-                             resource_name, username, password, mechanism):
-        # TODO: This should use the credentials passed via the constructor
-        # instead of accepting them as parameters
-        if resource == "topic":
-            resource = "--topic"
-        elif resource == "transactional-id":
-            resource = "--transactional-id"
+    def sasl_allow_principal(self,
+                             principal,
+                             operations,
+                             resource,
+                             resource_name,
+                             username: Optional[str] = None,
+                             password: Optional[str] = None,
+                             mechanism: Optional[str] = None):
+
+        username = username if username is not None else self._username
```

Comment:
maybe it would be a little confusing to allow overriding of the auth options?

Reply:
Yeah, just trying to avoid changing all the call sites. May as well do it though...I'll see how big a lift.

Reply:
Eh, kind of a large change, and some usage seems to override the provided creds with some intention. Will consider for a follow-on